bundles/org.openhab.binding.siemenshvac/src/main/java/org/openhab/binding/siemenshvac/internal/type/UidUtils.java

```diff
@@ -0,0 +1,217 @@
+            }
+            if (result.indexOf("lastd") >= 0) {
+                result = "holidays-hc-lastd";
+            }
+        }
+
+        result = result.replace("---", "-");
+        result = result.replace("--", "-");
+        result = result.replace('\'', '-');
+        result = result.replace('/', '-');
+        result = result.replace(' ', '-');
+        result = result.replace("+", "-");
+        result = result.replace("--", "-");
+
+        result = result.replace("standard-tsp-hc", "time-switch-program-standard");
+        result = result.replace("standard-tsp-4", "time-switch-program-standard");
+        result = result.replace("tsp-3", "time-switch-program-day");
+        result = result.replace("tsp-4", "time-switch-program-day");
+        result = result.replace("setpointtemp", "setpoint-temp-");
+        result = result.replace("rmtmp", "roomtemp");
+        result = result.replace("roomtempfrostprot", "room-temp-frostprot-");
+        result = result.replace("-setp", "-setpoint");
+        result = result.replace("optg", "operating-");
+        result = result.replace("-comf", "-comfort");
+        result = result.replace("-red", "-reduce");
+        result = result.replace("setp-", "-setpoint");
+        result = result.replace("roomtemp-", "room-temp-");
+        result = result.replace("-setpointhc", "-setpoint-hc");
+        result = result.replace("setphc", "-setpoint-hc");
+
+        return result;
+    }
+
+    /**
+     * Generates the ChannelTypeUID for the given datapoint with deviceType, channelNumber and datapointName.
+     */
+    public static ChannelTypeUID generateChannelTypeUID(SiemensHvacMetadataDataPoint dpt) throws Exception {
+
+        String type = dpt.getDptType();
+        String shortDesc = dpt.getShortDescEn();
+        String result = normalizeDescriptor(shortDesc);
+
+        try {
+            TypeConverter tp = ConverterFactory.getConverter(type);
+            if (!tp.hasVariant()) {
+                result = tp.getChannelType(dpt.getWriteAccess());
+            }
+        } catch (ConverterTypeException ex) {
+            throw new SiemensHvacException(String.format("Can't find convertor for type: %s", type), ex);
+        }
+
+        return new ChannelTypeUID(SiemensHvacBindingConstants.BINDING_ID, result);
+    }
+
+    public static ChannelTypeUID generateChannelTypeUID2(SiemensHvacMetadataDataPoint dpt) throws Exception {
+        String type = dpt.getDptType();
+        String shortDesc = dpt.getShortDescEn();
+        String result = normalizeDescriptor(shortDesc);
+
+        if ("DateTime".equals(type)) {
+            result = "datetime";
+        } else if ("String".equals(type)) {
+            result = "string";
+        } else if ("TimeOfDay".equals(type)) {
+            result = "datetime";
+        } else if ("Scheduler".equals(type)) {
+            result = "datetime";
+        }
+
+        return new ChannelTypeUID(SiemensHvacBindingConstants.BINDING_ID, result);
```

Comment:
Perhaps this can increase readability (and performance by not throwing away unneeded `normalizeDescriptor` result):
```suggestion
        String id = switch (dpt.getDptType()) {
            case "String" -> "string";
            case "DateTime", "TimeOfDay", "Scheduler" -> "datetime";
            default -> normalizeDescriptor(dpt.getShortDescEn());
        };

        return new ChannelTypeUID(SiemensHvacBindingConstants.BINDING_ID, id);
```